cores/nRF5/Tone.cpp

```diff
@@ -34,161 +34,397 @@ Version Modified By Date     Comments
 0010	Arduino.org 16/07/27 Added Arduino Primo support
 *************************************************/
 
-
+#include "Arduino.h"
 #include "Tone.h"
 #include "WVariant.h"
+#include "limits.h"
+
+// NOTE: Currently hard-coded to only use PWM2 ...
+//       These are the relvant hard-coded variables
+// TODO: Consider allowing dynamic use of any available PWM peripheral (but start with #2 for compatibility)
+static NRF_PWM_Type * const _PWMInstance = NRF_PWM2;
+static HardwarePWM  * const _HwPWM       = HwPWMx[2];
+
+// Defined a struct, to simplify validation testing ... also provides context when debugging
+class TONE_PWM_CONFIG {
+    private:
+        const nrf_pwm_clk_t      clock_frequency = NRF_PWM_CLK_125kHz;
+        const nrf_pwm_mode_t     pwm_mode        = NRF_PWM_MODE_UP;
+        const nrf_pwm_dec_step_t step_mode       = NRF_PWM_STEP_AUTO;
+        const nrf_pwm_dec_load_t load_mode       = NRF_PWM_LOAD_COMMON;
+        const uintptr_t          toneToken       = 0x656e6f54; //< 'T' 'o' 'n' 'e'
+    private:
+        uint64_t pulse_count;         //< total number of PWM pulses
+        uint32_t seq0_refresh;        //< count of pulses for each SEQ0 iteration
+        uint32_t seq1_refresh;        //< count of pulses for each SEQ1 iteration
+        uint16_t loop_count;          //< how many times to restart SEQ0/SEQ1?
+        uint16_t time_period;         //< how many clock cycles allocated to each PWM pulse?
+        uint16_t duty_with_polarity;  //< SEQ[N].PTR will point here, length == 1
+        uint8_t arduino_pin;          //< the arduino pin for playback
+        uint8_t nrf_pin;              //< the nrf pin for playback
+        nrf_pwm_task_t task_to_start; //< Whether to start playback at SEQ0 or SEQ1
+        nrf_pwm_short_mask_t shorts;  //< shortcuts to enable
+        boolean is_initialized;       //< defaults to uninitialized
+
+    public:
+        bool EnsurePwmPeripheralOwnership(void) noexcept;
+        bool InitializeFromPulseCountAndTimePeriod(uint64_t pulse_count, uint16_t time_period) noexcept;
```

Comment:
I think you may have the meaning backwards?  Having `noexcept` means [it will not throw any exception](https://en.cppreference.com/w/cpp/language/noexcept_spec).

Does this have any impact whatsoever on the code when compiled for the nRF52 platform as configured by Adafruit's BSP?  No.  None whatsoever.

But, in the spirit of showing flexibility, because you are uncomfortable with its existing, I've removed them.